lib/livebook_web/live/session_live/persistence_component.ex

```diff
@@ -46,14 +46,27 @@ defmodule LivebookWeb.SessionLive.PersistenceComponent do
       autosave_interval_s: autosave_interval_s
     }
 
+    draft_file =
+      cond do
+        file ->
+          file
+
+        match?({:file, _}, assigns.session.origin) ->
+          # if it's a forked notebook, use the same folder
+          assigns.session.origin
+          |> elem(1)
+          |> FileSystem.File.containing_dir()
+
+        true ->
+          Livebook.Config.local_file_system_home()
+      end
```

Comment:
You can do this:

```elixir
  file ||
    case assigns.session.origin do
      # If it's a forked notebook, default to the same folder
      {:file, file} -> FileSystem.File.containing_dir(file)
      _ -> Livebook.Config.local_file_system_home()
    end
```

Also, let's move it in the `:draft_file` function, so we only run this if needed :)